File: abelkhan_type_csharp_component_msgpack/component/center/hub_msg_handle.cs
/*
 * hub_msg_handle
 * 2020/6/2
 * qianqians
 */

namespace abelkhan
{
    public class hub_msg_handle
    {
        private hub_call_center_module _hub_call_center_module;
        private svrmanager svrmng;
        private hubmanager hubmng;
        private closehandle _closeHandle;

        public hub_msg_handle(modulemng modules, svrmanager svrs, hubmanager hubs, closehandle _closehandle)
        {
            svrmng = svrs;
            hubmng = hubs;
            _closeHandle = _closehandle;
            _hub_call_center_module = new hub_call_center_module(modules);

            _hub_call_center_module.onclosed += closed;
        }

        private void closed()
        {
            hubmng.hub_closed(_hub_call_center_module.current_ch);
            if (hubmng.check_all_hub_closed())
            {
                svrmng.close_db();
                _closeHandle.is_close = true;
            }
        }
    }
}

File: abelkhan_type_component/component/center/hub_msg_handle.ts
import abelkhan = require('../../abelkhan_type/ts/abelkhan');
import center_protcol = require('../../protcol/ts/center');
import svr = require('./svrmanager');
import hub = require('./hubmanager');
import close_handle = require('./closehandle');

export class hub_msg_handle{
    private hub_call_center_module : center_protcol.hub_call_center_module;
    private svrmng : svr.svrmanager;
    private hubmng : hub.hubmanager;
    private close_handle : close_handle.closehandle;
    constructor(modules:abelkhan.modulemng, svrs:svr.svrmanager, hubs:hub.hubmanager, _close_handle:close_handle.closehandle){
        this.hub_call_center_module = new center_protcol.hub_call_center_module(modules);
        this.svrmng = svrs;
        this.hubmng = hubs;
        this.close_handle = _close_handle;

        this.hub_call_center_module.cb_closed = this.closed.bind(this);
    }

    public closed(){
        this.hubmng.hub_closed(this.hub_call_center_module.current_ch);
        if (this.hubmng.check_all_hub_closed()){
            this.svrmng.close_db();
            this.close_handle.is_close = true;
        }
    }
}
